src/jaegertracing/SpanContext.h

```diff
@@ -160,9 +166,17 @@ class SpanContext : public opentracing::SpanContext {
 
     bool operator==(const SpanContext& rhs) const
     {
+        {
+            std::lock(_mutex, rhs._mutex);
+            std::lock_guard<std::mutex> lock(_mutex, std::adopt_lock);
```

Comment:
can this deadlock? this grabs its own lock and rhs grabs its own lock and then we're stuck?

Reply:
I was hoping you wouldn't pay too much attention to this. `std::lock` is a cool trick in C++11 that locks two mutexes without causing deadlock (essentially a loop of `try_lock` under the hood):

> Locks the given Lockable objects lock1, lock2, ..., lockn using a deadlock avoidance algorithm to avoid deadlock.
http://en.cppreference.com/w/cpp/thread/lock

 The lock guard stuff is just to make sure we unlock, doesn't acquire the lock again. Serves same purpose as `defer mutex.Unlock()` would in Go.